docker/dev-builder/android/Dockerfile

```diff
@@ -0,0 +1,41 @@
+FROM saschpe/android-ndk:34-jdk17.0.8_7-ndk25.2.9519653-cmake3.22.1
+
+ENV LANG en_US.utf8
+WORKDIR /greptimedb
+
+# Rename libunwind to libgcc
+RUN cp ${NDK_ROOT}/toolchains/llvm/prebuilt/linux-x86_64/lib64/clang/14.0.7/lib/linux/aarch64/libunwind.a ${NDK_ROOT}/toolchains/llvm/prebuilt/linux-x86_64/lib64/clang/14.0.7/lib/linux/aarch64/libgcc.a
+
+# Install dependencies.
+RUN apt-get update && apt-get install -y \
+    libssl-dev \
+    protobuf-compiler \
+    curl \
+    git \
+    build-essential \
+    pkg-config \
+    python3 \
+    python3-dev \
+    python3-pip \
+    && pip3 install --upgrade pip \
+    && pip3 install pyarrow
+
+# Trust workdir
+RUN git config --global --add safe.directory /greptimedb
+
+# Install Rust.
+SHELL ["/bin/bash", "-c"]
+RUN curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh -s -- --no-modify-path --default-toolchain none -y
+ENV PATH /root/.cargo/bin/:$PATH
+
+# Add android toolchains
+ARG RUST_TOOLCHAIN
+RUN rustup toolchain install ${RUST_TOOLCHAIN}
+RUN rustup target add aarch64-linux-android
+
+# Install cargo-ndk
+RUN cargo install cargo-ndk
+ENV ANDROID_NDK_HOME $NDK_ROOT
+
+# Builder entrypoint.
+CMD ["cargo", "ndk", "--platform", "23", "-t", "aarch64-linux-android", "build", "--bin", "greptime", "--profile", "release", "--no-default-features"]
```

Comment:
suggestion: We should put the build commands in Makefile to make it more flexible, for example:

```makefile
...
CARGO_NDK_OPTS ?= ndk --platform 23 -t aarch64-linux-android
...

ifneq ($(strip $(CARGO_BUILD_EXTRA_OPTS)),)
	CARGO_BUILD_OPTS += ${CARGO_BUILD_EXTRA_OPTS}
endif

.POHNY: build-by-dev-builder
build-by-dev-builder: ## Build greptime by dev-builder.
	docker run --network=host \
	-v ${PWD}:/greptimedb -v ${CARGO_REGISTRY_CACHE}:/root/.cargo/registry \
	-w /greptimedb ${IMAGE_REGISTRY}/${IMAGE_NAMESPACE}/dev-builder-${BASE_IMAGE}:latest \
	make build \
	CARGO_EXTENTION="${CARGO_EXTENTION}" \
	CARGO_PROFILE="${CARGO_PROFILE}" \
	FEATURES="${FEATURES}" \
	TARGET_DIR="${TARGET_DIR}" \
	TARGET="${TARGET}" \
	RELEASE="${RELEASE}" \
	CARGO_BUILD_EXTRA_OPTS="${CARGO_BUILD_EXTRA_OPTS}"
...
.PHONY: build-android-bin
build-android-bin: ## Build greptime binary for android.
	docker run --network=host \
	-v ${PWD}:/greptimedb -v ${CARGO_REGISTRY_CACHE}:/root/.cargo/registry \
	-w /greptimedb ${IMAGE_REGISTRY}/${IMAGE_NAMESPACE}/dev-builder-android:latest \
	make build \
	CARGO_EXTENTION="${CARGO_NDK_OPTS}" \
	CARGO_PROFILE=release \
	FEATURES="${FEATURES}" \
	TARGET_DIR="${TARGET_DIR}" \
	TARGET="${TARGET}" \
	RELEASE="${RELEASE}" \
	CARGO_BUILD_EXTRA_OPTS="--bin greptime --no-default-features"
...
```

Users can use the following makefile commands to build android binary:

```
make build-android-bin
```

The actual building commands will be:

```
cargo ndk --platform 23 -t aarch64-linux-android build --locked --profile release --bin greptime --no-default-features
```

Reply:
> suggestion: We should put the build commands in Makefile to make it more flexible, for example:
> 
> ```makefile
> ...
> CARGO_NDK_OPTS ?= ndk --platform 23 -t aarch64-linux-android
> ...
> 
> ifneq ($(strip $(CARGO_BUILD_EXTRA_OPTS)),)
> 	CARGO_BUILD_OPTS += ${CARGO_BUILD_EXTRA_OPTS}
> endif
> 
> ...
> .PHONY: build
> build: ## Build debug version greptime.
> 	cargo ${CARGO_EXTENTION} build ${CARGO_BUILD_OPTS}
> ...
> 
> .POHNY: build-by-dev-builder
> build-by-dev-builder: ## Build greptime by dev-builder.
> 	docker run --network=host \
> 	-v ${PWD}:/greptimedb -v ${CARGO_REGISTRY_CACHE}:/root/.cargo/registry \
> 	-w /greptimedb ${IMAGE_REGISTRY}/${IMAGE_NAMESPACE}/dev-builder-${BASE_IMAGE}:latest \
> 	make build \
> 	CARGO_EXTENTION="${CARGO_EXTENTION}" \
> 	CARGO_PROFILE="${CARGO_PROFILE}" \
> 	FEATURES="${FEATURES}" \
> 	TARGET_DIR="${TARGET_DIR}" \
> 	TARGET="${TARGET}" \
> 	RELEASE="${RELEASE}" \
> 	CARGO_BUILD_EXTRA_OPTS="${CARGO_BUILD_EXTRA_OPTS}"
> ...
> .PHONY: build-android-bin
> build-android-bin: ## Build greptime binary for android.
> 	docker run --network=host \
> 	-v ${PWD}:/greptimedb -v ${CARGO_REGISTRY_CACHE}:/root/.cargo/registry \
> 	-w /greptimedb ${IMAGE_REGISTRY}/${IMAGE_NAMESPACE}/dev-builder-android:latest \
> 	make build \
> 	CARGO_EXTENTION="${CARGO_NDK_OPTS}" \
> 	CARGO_PROFILE=release \
> 	FEATURES="${FEATURES}" \
> 	TARGET_DIR="${TARGET_DIR}" \
> 	TARGET="${TARGET}" \
> 	RELEASE="${RELEASE}" \
> 	CARGO_BUILD_EXTRA_OPTS="--bin greptime --no-default-features"
> ...
> ```
> 
> Users can use the following makefile commands to build android binary:
> 
> ```
> make build-android-bin
> ```
> 
> The actual building commands will be:
> 
> ```
> cargo ndk --platform 23 -t aarch64-linux-android build --locked --profile release --bin greptime --no-default-features
> ```

What should I do if I also want to invoke strip command inside builder container to strip the binary?

Reply:
@v0y4g3r @sunng87 Maybe using [`nightly`](https://github.com/GreptimeTeam/greptimedb/blob/develop/Cargo.toml#L175-L180) is OK? If not, we can define the new profile as `nigtly`.

Reply:
> suggestion: We should put the build commands in Makefile to make it more flexible, for example:
> 
> ```makefile
> ...
> CARGO_NDK_OPTS ?= ndk --platform 23 -t aarch64-linux-android
> ...
> 
> ifneq ($(strip $(CARGO_BUILD_EXTRA_OPTS)),)
> 	CARGO_BUILD_OPTS += ${CARGO_BUILD_EXTRA_OPTS}
> endif
> 
> ...
> .PHONY: build
> build: ## Build debug version greptime.
> 	cargo ${CARGO_EXTENTION} build ${CARGO_BUILD_OPTS}
> ...
> 
> .POHNY: build-by-dev-builder
> build-by-dev-builder: ## Build greptime by dev-builder.
> 	docker run --network=host \
> 	-v ${PWD}:/greptimedb -v ${CARGO_REGISTRY_CACHE}:/root/.cargo/registry \
> 	-w /greptimedb ${IMAGE_REGISTRY}/${IMAGE_NAMESPACE}/dev-builder-${BASE_IMAGE}:latest \
> 	make build \
> 	CARGO_EXTENTION="${CARGO_EXTENTION}" \
> 	CARGO_PROFILE="${CARGO_PROFILE}" \
> 	FEATURES="${FEATURES}" \
> 	TARGET_DIR="${TARGET_DIR}" \
> 	TARGET="${TARGET}" \
> 	RELEASE="${RELEASE}" \
> 	CARGO_BUILD_EXTRA_OPTS="${CARGO_BUILD_EXTRA_OPTS}"
> ...
> .PHONY: build-android-bin
> build-android-bin: ## Build greptime binary for android.
> 	docker run --network=host \
> 	-v ${PWD}:/greptimedb -v ${CARGO_REGISTRY_CACHE}:/root/.cargo/registry \
> 	-w /greptimedb ${IMAGE_REGISTRY}/${IMAGE_NAMESPACE}/dev-builder-android:latest \
> 	make build \
> 	CARGO_EXTENTION="${CARGO_NDK_OPTS}" \
> 	CARGO_PROFILE=release \
> 	FEATURES="${FEATURES}" \
> 	TARGET_DIR="${TARGET_DIR}" \
> 	TARGET="${TARGET}" \
> 	RELEASE="${RELEASE}" \
> 	CARGO_BUILD_EXTRA_OPTS="--bin greptime --no-default-features"
> ...
> ```
> 
> Users can use the following makefile commands to build android binary:
> 
> ```
> make build-android-bin
> ```
> 
> The actual building commands will be:
> 
> ```
> cargo ndk --platform 23 -t aarch64-linux-android build --locked --profile release --bin greptime --no-default-features
> ```

Fixed in https://github.com/GreptimeTeam/greptimedb/pull/2541/commits/7b5d3373658638d14c81419ede0cab9b1cbf8387#diff-76ed074a9305c04054cdebb9e9aad2d818052b07091de1f20cad0bbac34ffb52